File: tests/benchmarks/cpu_bench.cpp
#include <iostream>
#include <cmath>
#include <chrono>
#include <iomanip>

int main() {
    int n = 10000000;
    auto start = std::chrono::high_resolution_clock::now();
    double total = 0.0;
    for (int i = 1; i <= n; ++i) {
        total += std::sqrt(i) * std::sin(i);
    }
    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> diff = end - start;
    std::cout << "Result: " << std::fixed << std::setprecision(4) << total << std::endl;
    std::cout << "CPU Test: " << std::fixed << std::setprecision(4) << diff.count() << " sec" << std::endl;
    return 0;
}

File: tests/benchmarks/cpu_bench.rs
use std::time::Instant;

fn main() {
    let n = 10_000_000;
    let start = Instant::now();
    let mut total: f64 = 0.0;
    for i in 1..=n {
        total += (i as f64).sqrt() * (i as f64).sin();
    }
    let duration = start.elapsed();
    println!("Result: {:.4}", total);
    println!("CPU Test: {:.4} sec", duration.as_secs_f64());
}
